src/main/java/com/thebuzzmedia/exiftool/core/UnspecifiedTag.java

```diff
@@ -0,0 +1,63 @@
+package com.thebuzzmedia.exiftool.core;
+
+import com.thebuzzmedia.exiftool.Tag;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import static com.thebuzzmedia.exiftool.Constants.SEPARATOR;
+
+/**
+ * Class that can represent any tag, not just the standard tags.
+ * <p>
+ * Since the type of value is unknown, typed parsing is not possible,
+ * and the tag does not know whether multiple values are expected or not.
+ * Consequently, when parsing, it returns a String[] regardless of whether
+ * the tag will ever have multiple values. Further parsing is then up to the caller.
+ *
+ * @author David Edwards (david@more.fool.me.uk)
+ */
+
+public class UnspecifiedTag implements Tag {
+
+	private String name;
+
+	public UnspecifiedTag(String name) {
+		this.name = name;
+	}
+
+	@Override
+	public String getName() {
+		return name;
+	}
+
+	@Override
+	public String getDisplayName() {
+		return name;
+	}
+
+	@Override
+	public <T> T parse(String value) {
+		// Clearly this won't work if caller is expecting anything other than
+		// a string array. This is due to the way the {@link Tag} interface is designed.
```

Comment:
> This is due to the way the {@link Tag} interface is designed.

Do you have any suggestion to improve this new feature?
I would be happy to hear any suggestions/review all the changes you think could be useful, feel free to respond or submit another issue/PR ;)

Reply:
I think the problem with this interface is the use of method generics rather than class generics. By specifying <T> on the method, you are saying that the calling code specifies the type it wants, even though the parse methods on the tag implementations only support specific conversions. This is a problem with the existing StandardTag class too.

For example, if I ask for the ISO tag, which parses to an Integer, I can still write:

    WhateverClass value = ISO.parse("100")

and it will compile just fine, but throw a class cast exception when you run it.

If you made the Tag interface generic (ie, Tag&lt;T&gt;), and remove the &lt;T&gt; on the parse method then the return type of the `parse` method can be checked at compile time and the unsafe class casts can be removed. However, you'd have to convert the `StandardTag` enum into a class. That might not be a bad thing, though. You could still keep the constructor private if you want, and define the tags as static final class instances instead:

    public static final ISO = new StandardTag<Integer>( ... )

Which would allow callers to refer to them in exactly the same way as currently, but with compile-time type safety.